jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -42,3 +48,97 @@ def on_mouse_event(self, data):
         self.viewer.session.hub.broadcast(msg)
 
         self._time_last = time.time()
+
+
+@viewer_tool
+class SpectrumPerSpaxel(CheckableTool):
+
+    icon = os.path.join(ICON_DIR, 'contrast.svg')
+    tool_id = 'jdaviz:spectrumperspaxel'
+    action_text = 'See spectrum at a single spaxel'
+    tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
+
+    def __init__(self, viewer, **kwargs):
+        self.old_y_min = 0
+        self.old_y_max = 0
+        self.spectrum_viewer = None
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_or_key_event,
+                                       events=['click'])
+        self.spectrum_viewer = self.viewer.jdaviz_app.get_viewer_by_id("cubeviz-3")
+        self.old_y_min, self.old_y_max = (self.spectrum_viewer.state.y_min,
+                                          self.spectrum_viewer.state.y_max)
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_or_key_event)
+        with delay_callback(self.spectrum_viewer.state, 'y_min', 'y_max'):
+            self.spectrum_viewer.state.y_min, self.spectrum_viewer.state.y_max = (self.old_y_min,
+                                                                                  self.old_y_max)
+
+    def on_mouse_or_key_event(self, data):
+        event = data['event']
+
+        if event == 'click':
+            event_x = data['domain']['x']
+            event_y = data['domain']['y']
+
+            x = round(event_x)
+            y = round(event_y)
+
+            new_y_min = 0
+            new_y_max = 0
+            for index, data in enumerate(self.viewer.data()):
+                if len(data.shape) != 3:
+                    return
+
+                cube = data.get_object(cls=Spectrum1D, statistic=None)
+                if x > cube.shape[0] or y > cube.shape[1] or x < 0 or y < 0:
+                    return
+
+                # print(f"{data.label} at {x}, {y}")
+
+                # TODO: Remove failed attempt at wcs setting
+                # wcs = WCS(naxis=1)
+                # wcs.wcs.ctype = [cube.wcs.wcs.ctype[0]]
+                # wcs.wcs.cunit = [cube.wcs.wcs.cunit[0]]
+                # wcs.wcs.set()
+
+                spec = Spectrum1D(flux=cube.flux[x][y], spectral_axis=cube.spectral_axis)
+                label = f"{data.label}_at_pixel"
+                self.viewer.jdaviz_app.remove_data_from_viewer("spectrum-viewer", label)
+                # self.viewer.session.data_collection[label] = spec
+                self.viewer.jdaviz_app.add_data(spec, label)
+                self.viewer.jdaviz_app.add_data_to_viewer("spectrum-viewer", label, clear_other_data=False)
```

Comment:
I wonder if glue Data has API to just update the underlying data array rather than completely remove and re-add. 🤔

Reply:
I'll look into it, that would definitely be helpful!